### PR TITLE
Change attention mask dtype to be bool. Fix #1119

### LINKED ISSUE #1119: Tons of warnings on use of TransfoXLModel. masked_fill_ input dtype torch.uint8 should be changed to torch.bool
## 🕑 Usage of deprecated behaviour

Using example from documentation web page: https://huggingface.co/pytorch-transformers/model_doc/transformerxl.html#pytorch_transformers.TransfoXLModel
```
import torch
from pytorch_transformers import *

tokenizer = TransfoXLTokenizer.from_pretrained('transfo-xl-wt103')
model = TransfoXLModel.from_pretrained('transfo-xl-wt103')
input_ids = torch.tensor(tokenizer.encode("Hello, my dog is cute")).unsqueeze(0)
outputs = model(input_ids)
last_hidden_states, mems = outputs[:2]
```

Get tons of same warnings:
> /pytorch/aten/src/ATen/native/LegacyDefinitions.cpp:14: UserWarning: masked_fill_ received a mask with dtype torch.uint8, this behavior is now deprecated,please use a mask with dtype torch.bool instead.
image

Created #1120 to fix it.

### PR DESCRIPTION
**Pull request by CrafterKolyan**

### CODE REVIEW

**Comment by thomwolf** [comment]
Yes it's better, thanks also for that @CrafterKolyan!
